js/web/lib/wasm/jsep/webgpu/ops/common.ts

```diff
@@ -805,4 +805,4 @@ export const getBroadcastDims = (inShape: readonly number[], outShape: readonly
 };
 
 // TODO: remove this limitation once >4D dims are supported by uniform.
-export const enableShapesUniforms = (rank: number): boolean => rank <= 4;
+export const enableShapesUniforms = (rank: number): boolean => rank <= 4 && rank > 0;
```

Comment:
Why `rank = 0` can't use uniform?

Reply:
Ah, ignore this comment. We don't need shapes info for scalar.

Reply:
Sorry for the back-and-forth. In binary-op.ts, you use `inputDependencies: useShapesUniforms ? ['rank', 'rank'] : ['dims', 'dims']`. In this case, if the input is scalar, `useShapesUniforms ` will be false which results the inputDependencies is `dims`. It's not what we want. We should make sure that` inputDependencies` depends on `type` or `rank` if the input is scalar.